lib/ea_restaurant_data_loader_web/controllers/oauth2_controller.ex

```diff
@@ -0,0 +1,13 @@
+defmodule EaRestaurantDataLoaderWeb.Controllers.Oauth2Controller do
+  use EaRestaurantDataLoaderWeb, :controller
+  alias EaRestaurantDataLoader.Lib.Utils.Oauth2Util
+  alias EaRestaurantDataLoader.Lib.Services.Oauth2Service
+
+  def login(conn, _params) do
+    [basic_auth | _] = conn |> get_req_header("authorization")
+    [client_id, client_secret] = Oauth2Util.get_client_credentials_from_basic_auth(basic_auth)
+    login_response = Oauth2Service.login_client(client_id, client_secret)
+    {_, login_json_response} = Poison.encode(login_response)
```

Comment:
```
{_, login_json_response} =
  Oauth2Service.login_client(client_id, client_secret)
    |> Poison.encode()
```

Reply:
Also could you create a function to convert an object to json? Just in case we need to extend the parsiong functionality later.

Could be something like `AppUtil.parse_to_json(...)`

Reply:
Done.